sync2/rangesync/rangesync.go

```diff
@@ -552,8 +555,11 @@ func (rsr *RangeSetReconciler) handleSample(
 	msg SyncMessage,
 	info RangeInfo,
 ) (pr ProbeResult, err error) {
-	pr.FP = msg.Fingerprint()
+	pr.InSync = msg.Fingerprint() == info.Fingerprint
 	pr.Count = msg.Count()
+	if pr.InSync && pr.Count != msg.Count() {
+		return ProbeResult{}, errors.New("mismatched count with matching fingerprint, collision?")
```

Comment:
NIT: errors shouldn't ask questions 😉 

```suggestion
		return ProbeResult{}, errors.New("mismatched count with matching fingerprint")
```

Reply:
Removed question but kept mention of possible collision to make a possible culprit more clear